lib/jira/commands/checkout.rb

```diff
@@ -0,0 +1,88 @@
+module Jira
+  class CLI < Thor
+
+    desc "checkout", "Checks out a ticket from JIRA in the git branch"
+    method_option :remote, aliases: "-r", type: :string, default: nil, lazy_default: "", banner: "REMOTE"
+    def checkout(ticket=Jira::Core.ticket)
+      Command::Checkout.new(ticket, options).run
+    end
+
+  end
+
+  module Command
+    class Checkout < Base
+
+      attr_accessor :ticket, :options
+
+      def initialize(ticket, options)
+        self.ticket = ticket
+        self.options = options
+      end
+
+      def run
+        return unless Jira::Core.ticket?(ticket)
+        return if metadata.empty?
+        unless metadata['errorMessages'].nil?
+          on_failure
+          return
+        end
+        unless remote?
+          on_failure
+          return
+        end
+
+        branches_include = branches.include?(ticket)
```

Comment:
probably a little cleaner to memoize `branches` and call the `branches.include?(ticket)` inline rather than storing a variable